gilmour.go

```diff
@@ -523,16 +536,14 @@ func isExpired(timeout time.Duration, startTime time.Time) bool {
 	return time.Now().After(startTime.Add(timeout))
 }
 
-func try(r RetryConf, fn func() error) (err error) {
+func try(g *Gilmour, fn func() error) (err error) {
 	startTime := time.Now()
 	for {
 		err = fn()
-		if isExpired(r.Timeout, startTime) || !isNetworkError(err) || err == nil {
+		if isExpired(g.getRetryTimeout(), startTime) || !isNetworkError(err) || err == nil {
```

Comment:
@mohandutt134 But most Network calls accept timeouts. Example:
You are making tea and need milk immediately. You send your brother to the mother dairy to fetch some milk, asking him to return if the queue has a wait period of longer than 10 minutes.
Scenario A: Would you want him to return after 10 minutes so that you can send him to another shop?
Scenario B: He keeps waiting forever in the queue and comes back 2 hours later and acknowledges that queue was longer than 10 minutes.

Which scenario is ideal?